orcid-core/src/main/java/org/orcid/core/manager/v3/read_only/impl/ProfileEntityManagerReadOnlyImpl.java

```diff
@@ -32,32 +30,25 @@ public Boolean isLocked(String orcid) {
     
     @Override
     public String getLockedReason(String orcid) {
-        List<Object[]> result = profileDao.getLockedReason(orcid);
-        String reason = (String) result.get(0)[0];
-        String description = (String) result.get(0)[1];
-        String adminUser = (String) result.get(0)[2];
+        ProfileEntity profileEntity = profileDao.getLockedReason(orcid);
+        String reason = profileEntity.getReasonLocked();
+        String description = "";
+        String adminUser = "";
         String strDate = "";
-        Date date = new Date();
 
         // format results
-        if (adminUser == null || adminUser.isEmpty()) {
-            adminUser = "";
-        } else {
-            adminUser = " by " + adminUser;
+        if (profileEntity.getRecordLockingAdmin() != null) {
+            adminUser = " by " + profileEntity.getRecordLockingAdmin();
         }
 
-        // If record_locked_date is missing, try to use last_indexed_date
-        if ((Date) result.get(0)[3] == null) {
-            date = (Date) result.get(0)[4];
-        } else if ((Date) result.get(0)[4] != null) {
-            SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy");
-            strDate = " on " + sdf.format(date);
-        } else
-            strDate = "";
+        SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy");
+        if (profileEntity.getRecordLockedDate() != null) {
+            strDate = " on " + sdf.format(profileEntity.getRecordLockedDate());
+        } else if (profileEntity.getLastIndexedDate() != null) {
```

Comment:
@DanielPalafox this should be getLastModifiedDate() instead of getLastIndexedDate() right?